src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T & string): GroupsMap<T> {
+  const groups: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const groupKey: T [keyof T & string] = item[key];
+
+    if (groups[groupKey as string]) {
+      groups[groupKey as string].push(item);
+    } else {
+      groups[groupKey as string] = [item];
+    }
```

Comment:
1. it's better to check an existing of the key in the object, not a value of the key
2. remove alias
```suggestion
    if (groupKey in groups) {
      groups[groupKey].push(item);
    } else {
      groups[groupKey] = [item];
    }
```